.github/workflows/tests.yml

```diff
@@ -0,0 +1,42 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  tests:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/setup.py') }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install codecov pytest
+        python3 -m pip install .[test]
```

Comment:
Can we use one consistent way to invoke `pip`, rather than 3 different ways? :wink: 

Pytest is one of our test dependencies - `.[test]` will include it, so we don't need to list it separately.

The [codecov Github action](https://github.com/codecov/codecov-action) doesn't use the Python codecov package, so I don't think we need to install that. (It, uh... [downloads a bash script to execute instead](https://github.com/codecov/codecov-action/blob/239febf655bba88b16ff5dea1d3135ea8663a1f9/index.js#L32-L45)... maybe it's nicer to use the Python package directly. :-S )

Reply:
Oops yes that shouldn't be there, I'll fix it now, thanks!